fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/RangeInference.java

```diff
@@ -130,10 +132,22 @@ private ValueDesc simplify(ExpressionRewriteContext context,
             Expression originExpr, List<Expression> predicates,
             BinaryOperator<ValueDesc> op, boolean isAnd) {
 
+        boolean convertIsNullToEmptyValue = isAnd && predicates.stream().anyMatch(expr -> expr instanceof NullLiteral);
```

Comment:
it is better to check if predicates contains "isNull" and "null" literal.
if match, return EmtpyValue directly, and skip the for loop in line138-150

Reply:
this pr only solved user case. other pr will fix this.